### PR TITLE
[GEOS-11520] Repository CITATION.cff metadata

### PR DESCRIPTION
**Pull request by jodygarnett**
[GEOS-11520](https://osgeo-org.atlassian.net/browse/GEOS-11520) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

Our existing `zenodo.json` has stopped working (sigh). This PR provides the recommended `CITATION.cff` file as a replacement.

You can edit using [cffinit tool](https://citation-file-format.github.io/cff-initializer-javascript/#/) if you like.

What is this stuff:
* https://citation-file-format.github.io/
* https://wiki.osgeo.org/wiki/DOI#DOI_available_for_OSGeo_Projects
* https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    city: Via Carignoni
+    country: IT
+    post-code: '55041'
+    address: Via Carignoni 51
+    location: Camaiore
+    region: Lucca
+    tel: '+390584962313'
+    fax: '+390584660272'
+    website: 'https://www.geosolutionsgroup.com/'
+    email: info@geosolutionsgroup.com
```
Comment (jodygarnett) [inline CITATION.cff:77]: @simboss do I have address information correct here?

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  - standards
+  - spatial
+  - mapping
+  - open geospatial consortium
+  - open source geospatial foundation
+  - osgeo
+license: GPL-2.0
+commit: 74f1315632575736f93c8e50e3997c8481e8f1aa
+version: 2.24.5
+date-released: '2024-08-26'
```
Comment (jodygarnett) [inline CITATION.cff:123]: It looks like this version specific stuff is an optional and can be removed

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,123 @@
+# This CITATION.cff file was generated with cffinit.
+# Visit https://bit.ly/cffinit to generate yours today!
+
+cff-version: 1.2.0
+title: GeoServer
+message: >-
+  If you use this software, please cite it using the
+  metadata from this file.
```
Comment (jodygarnett) [inline CITATION.cff:8]: this is the default message, I assume it is fine

**Review by mprins — APPROVED** [review]

**Comment by jodygarnett** [comment]
Okay this is just waiting on confirmation of GeoSolutions address. Perhaps someone more knowledge of Italy can check against https://www.geosolutionsgroup.com/about/

> GeoSolutions SRL
> Registered Office: Via Carignoni 51, 55041 Camaiore (Lucca)
> VAT Number and Tax Code: 02036020465 
> Phone:+390584962313
> Fax:+390584660272

I just do not know how to map the address fields :)

**Comment by simboss** [comment]
> Okay this is just waiting on confirmation of GeoSolutions address. Perhaps someone more knowledge of Italy can check against https://www.geosolutionsgroup.com/about/
> 
> > GeoSolutions SRL
> > Registered Office: Via Carignoni 51, 55041 Camaiore (Lucca)
> > VAT Number and Tax Code: 02036020465
> > Phone:+390584962313
> > Fax:+390584660272
> 
> I just do not know how to map the address fields :)

The info is correct @jodygarnett, what do you exactly need?

**Comment by jodygarnett** [comment]
Thanks @simboss I just wanted to confirm that I under understood the Italian address format well enough to fill in the fields.

**Review by ianturton — APPROVED** [review]

**Comment by ianturton** [comment]
merged via cli

**Comment by ianturton** [comment]
Wondering if we should do the same for GeoTools now

File: CITATION.cff (hunk 1 above)
Comment (jodygarnett) [reply CITATION.cff:80]: Okay got some clarification, "Via Carignoni" was not the city, it translates to "Carignoni Road":

Revised address should be:
```
  - name: GeoSolutions SRL
    city: Camaiore
    country: IT
    post-code: '55041'
    address: Via Carignoni 51
    location: Lucca
    region: Tuscany
```

@ianturton do you wish to update?

**Comment by jodygarnett** [comment]
@ianturton:

> merged via cli

I do not see the CITATION.cff file in the repository, and the GeoSolutions address above is incorrect. Please advise.

**Pull request merged**